GAE/src/com/gallatinsystems/framework/dao/BaseDAO.java

```diff
@@ -635,6 +637,25 @@ protected T findByProperty(String propertyName, Object propertyValue,
         pm.deletePersistentAll(obj);
     }
 
+    /**
+     * deletes an object from the db, by key
+     *
+     * @param <E>
+     * @param obj
+     */
+    public <E extends BaseDomain> void deleteByKey(Key k) {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        datastore.delete(k);
+    }
+
+    /**
+     * deletes a list of objects, by key, in a single datastore interaction
+     */
+    public <E extends BaseDomain> void deleteByKeys(Collection<Key> kList) {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        datastore.delete(kList);
```

Comment:
Would there be an issue if the number of objects is very large?

Reply:
That is why we delete by keys. They are much smaller than the entites, so hopefully we have memory for all of them. (Should only be an issue on the first run. and is harmless if it fails. If we see errors, we can revisit...)